sklearn/neighbors/lof.py

```diff
@@ -0,0 +1,265 @@
+# -*- coding: utf-8 -*-
+
+import numpy as np
+
+from .base import NeighborsBase
+from .base import KNeighborsMixin
+from .base import UnsupervisedMixin
+
+from ..metrics import pairwise_distances
+
+from ..utils import check_array
+from ..utils import _get_n_jobs
+
+__all__ = ["LOF"]
+
+
+class LOFMixin(object):
+    """Mixin for Local-Outlier-Factor computation"""
+
+    def k_distance(self, X=None):
+        """
+        Compute the k_distance and the neighborhood of querry samples X wrt
+        self._fit_X.
+        If X=None, neighbors of each sample self._fit_X are returned.
+        In this case, the query point is not considered its own neighbor.
+        """
+        distances, neighbors_indices = self.kneighbors(
+            X=X, n_neighbors=self.n_neighbors)
+        neighbors_indices = neighbors_indices
+        k_dist = distances[:, self.n_neighbors-1]
+
+        return k_dist, neighbors_indices
+
+    def local_reachability_density(self, p):
+        """The local reachability density (LRD) of p is the inverse of the
+        average reachability distance based on the self.n_neighbors-nearest
+        neighbors of instance.
+
+        Parameters
+        ----------
+
+        p : array-like of shape (n_samples, n_features)
+        The samples to compute the LRD w.r.t. self._fit_X
+        If None, compute the LRD of self._fit_X w.r.t. to itself.
+        (In this case samples are not considered in their own neighbourhood)
+
+        Returns
+        -------
+        local reachability density : float
+        The LRD of p.
+        """
+
+        p_0 = self._fit_X if p is None else p
+
+        neighbors_indices = self.neighbors_indices_fit_X_ if p is None else self.k_distance(p)[1]
+
+        n_jobs = _get_n_jobs(self.n_jobs)
```

Comment:
When you merge this into the LOF class, just call _get_n_jobs once in the `__init__` function.